tiledb/sm/query/query.cc

```diff
@@ -944,387 +910,97 @@ void Query::set_processed_conditions(
-  // behavior before subarray was exposed directly
-  subarray_.set_config(config_);
-}
-
-Status Query::set_coords_buffer(void* buffer, uint64_t* buffer_size) {
-  // Set zipped coordinates buffer
-  coords_info_.coords_buffer_ = buffer;
-  coords_info_.coords_buffer_size_ = buffer_size;
-  coords_info_.has_coords_ = true;
-
-  return Status::Ok();
-}
-
-void Query::set_dimension_label_buffer(
-    const std::string& name, const QueryBuffer& buffer) {
-  if (buffer.buffer_var_ || buffer.buffer_var_size_) {
-    // Variable-length buffer. Set data buffer and offsets buffer.
-    throw_if_not_ok(
-        set_data_buffer(name, buffer.buffer_var_, buffer.buffer_var_size_));
-    throw_if_not_ok(set_offsets_buffer(
-        name, static_cast<uint64_t*>(buffer.buffer_), buffer.buffer_size_));
-  } else {
-    // Fixed-length buffer. Set data buffer only.
-    throw_if_not_ok(set_data_buffer(name, buffer.buffer_, buffer.buffer_size_));
-  }
-}
-
-Status Query::set_data_buffer(
-    const std::string& name,
-    void* const buffer,
-    uint64_t* const buffer_size,
-    const bool check_null_buffers,
-    const bool serialization_allow_new_attr) {
-  // General checks for fixed buffers
-  RETURN_NOT_OK(check_set_fixed_buffer(name));
-
-  // Check buffer
-  if (check_null_buffers && buffer == nullptr) {
-    if ((type_ != QueryType::WRITE && type_ != QueryType::MODIFY_EXCLUSIVE) ||
-        *buffer_size != 0) {
-      return logger_->status(
-          Status_QueryError("Cannot set buffer; " + name + " buffer is null"));
-    }
-  }
-
-  // Check buffer size
-  if (check_null_buffers && buffer_size == nullptr) {
-    return logger_->status(Status_QueryError(
-        "Cannot set buffer; " + name + " buffer size is null"));
-  }
-
-  // If this is for a dimension label, set the dimension label buffer and
-  // return.
-  if (array_schema_->is_dim_label(name)) {
-    // Check the query type is valid.
-    if (type_ != QueryType::READ && type_ != QueryType::WRITE) {
-      throw QueryStatusException("[set_data_buffer] Unsupported query type.");
-    }
-
-    // Set dimension label buffer on the appropriate buffer depending if the
-    // label is fixed or variable length.
-    array_schema_->dimension_label(name).is_var() ?
-        label_buffers_[name].set_data_var_buffer(buffer, buffer_size) :
-        label_buffers_[name].set_data_buffer(buffer, buffer_size);
-    return Status::Ok();
-  }
+    // Set dimension label buffer on the appropriate buffer depending if the
+    // label is fixed or variable length.
+    array_schema_->dimension_label(name).is_var() ?
+        label_buffers_[name].set_data_var_buffer(buffer, buffer_size) :
+        label_buffers_[name].set_data_buffer(buffer, buffer_size);
+    return Status::Ok();
+  }
 
   // For easy reference
   const bool is_dim = array_schema_->is_dim(name);
   const bool is_attr = array_schema_->is_attr(name);
+  const bool is_aggr = is_aggregate(name);
```

Comment:
As noted earlier, what happens if both `is_attr` and `is_aggr` are both true. And for that matter `is_dim`? I suddenly have no idea what happens if a user creates an attribute and dimension with the same name.

Reply:
This will need to be prevented at the API layer. Names should be unique across dimensions/attributes and aggregate output field names.